api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
I guess the question is what was the behavior after your change? 
I.e if you void and unvoid an Obs with no other changes in both operations, do you shouldn't get an extra 'replacment' obs for the old one, but if there in any other changes then it should get voided and replaced with a new one. The tricky thing is how to know if there were changes when it was initially getting voided. This is why one might argue that may be we should have your change, I guess we need to know what the behavior is.